test-data/unit/check-classes.test

```diff
@@ -6077,3 +6077,15 @@ TX = TypeVar('TX', bound='X')
 class X:
     def __new__(cls, x: TX) -> TX:  # E: "__new__" must return a class instance (got "TX")
         pass
+
+[case testGenericOverride]
+from typing import Generic, TypeVar, Any
+
+T = TypeVar('T')
+
+class B(Generic[T]):
+    x: T
+
+class C(B):
+    def __init__(self) -> None:
+        self.x: Any
```

Comment:
Hm, I am not super happy with the error message, because it says:
```
Incompatible types in assignment (expression has type "int", base class "B" defined the type as "str")
```
while the initial type is `T` (it shows the mapped/expanded type). It is however not easy to improve, so I think it is OK.